common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -195,15 +205,15 @@ private static int numBytesForFirstByte(final byte b) {
    * Returns the number of bytes
    */
   public int numBytes() {
-    return numBytes;
+    return (int)base.size();
```

Comment:
shall we check overflow before cast?

Reply:
ah now I see the point of having `UTF8String.numBytes`. `MemoryBlock.size` is long and here we need a int, and `numBytes()` is called many times so performance matters.

Reply:
Yeah, sure

Reply:
Yes, I believe that `UTF8String` always puts int value into `MemoryBlock.size`. However, some corruption may occur. Let me add assertion for testing.

Reply:
Finally, I removed assertion due to performance reason.

Reply:
shall we just create a `int numBytes` in `UTF8String` for performance? the type cast may also hurt performance.